Agency Dashboard SMS Notification display contact numbers

TheSteveK opened the pull request:
Related to # 1204774821045518-as-1204784983353063/f

## Proposed Changes

This PR supersedes https://github.com/Automattic/wp-calypso/pull/77965

* This PR adds a new component to the SMS section in the settings modal
* The component displays all of the SMS numbers and names, similar to the email field
* I've also added the context menu for additional numbers, but none of the items are functional

## Testing Instructions

![Screenshot 2023-06-08 at 10 27 15 AM](https://github.com/Automattic/wp-calypso/assets/12895386/0d3f8b98-1765-434b-b2ca-607da92861cd)

![Screenshot 2023-06-08 at 10 27 40 AM](https://github.com/Automattic/wp-calypso/assets/12895386/7d2e5e58-394a-47ad-942c-9d24ca6c5263)

* checkout this PR locally and fire up yarn start-jetpack-cloud
* Make sure you have an agency partner account and head over to jetpack.cloud.localhost:3000 
* Click the notification settings icon next to one of your sites
* If it's not toggled on, toggle on SMS notification
* Click the add button to add a phone number
* Click verify and then click later to add it to the list
* You should now see your newly added phone number in the modal
*  Click the elipse to the right of the 2nd contact to see the context menu
* Enable email to verify that it still correctly shows email contacts for the site

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

TheSteveK committed:
Downtime Montoring SMS add phone number list to modal

TheSteveK committed:
remove default sms move sms type to types.ts

TheSteveK committed:
unify css

TheSteveK committed:
fix paths, imports and alert banner

TheSteveK committed:
remove email specific stylesheet

TheSteveK committed:
fix typescript error

mattgawarecki commented on client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/configure-sms-notification/sms-item-content.tsx:
```diff
@@ -0,0 +1,124 @@
+	const buttonActionRef = useRef< HTMLButtonElement | null >( null );
+
+	const showActions = () => {
+		setIsOpen( true );
```
When opening the action menu, I noticed that to close it I had to click away from the menu -- clicking the button again did nothing. Just checking, is that our desired behavior? Intuitively I would expect to be able to close the menu by clicking the button again, sort of like a toggle.

mattgawarecki commented on client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/configure-sms-notification/sms-item-content.tsx:
```diff
@@ -0,0 +1,124 @@
+					<div className="configure-contact-info__card-sub-heading">{ item.name }</div>
+				</span>
+
+				<>
```
This enclosing fragment (i.e., `<>`) isn't strictly necessary, since the parent `div` supports multiple children.

mattgawarecki commented on client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/style.scss:
```diff
@@ -13,7 +13,80 @@ button.configure-contact__button {
 }
 
-.configure-contact__form {
+$help-text-color: #757575;
```
Is this color variable used anywhere?

mattgawarecki commented on client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/style.scss:
```diff
@@ -13,7 +13,80 @@ button.configure-contact__button {
+	height: 60px;
+	display: flex;
+	align-items: center;
+	padding: 16px !important;
```
Main comment to address a few examples in this same file:

Let's avoid the use of `!important` if at all possible -- it makes overriding these rules in the future very confusing/difficult.

An alternate approach would be to add more specificity to this rule set, like specifying another class, ID, or element. In a pinch, when there's no other identifier to apply, you can even double up on the same one (e.g., `.configure-contact-info__card.configure-content-info__card`).

mattgawarecki reviewed (commented):
Functionally the behavior in this PR seems _mostly_ right, but I did leave a question and a few comments to address before deploying. Nice work, especially incorporating feedback from your earlier PR!

TheSteveK committed:
make slider button close settings

jkguidaven commented:
@TheSteveK can you also rebase this to the latest trunk so it pickups the latest changes on the verify modal 🙏. 

My [PR](https://github.com/Automattic/wp-calypso/pull/78101) is based on this branch and also requires the changes on the trunks to properly test.

TheSteveK commented:
> @TheSteveK can you also rebase this to the latest trunk so it pickups the latest changes on the verify modal 🙏.
> 
> My [PR](https://github.com/Automattic/wp-calypso/pull/78101) is based on this branch and also requires the changes on the trunk to properly test.

Done!

TheSteveK committed:
improve css specificity

TheSteveK commented:
@mattgawarecki thanks for the comments I've addressed them.  Since @yashwin did the original CSS work, I'll ping him here to let me know if there's a specific reason for the !important directives I don't know about, but I fixed those by nesting the affected scss for better specificity.

mattgawarecki commented:
I re-tested and re-examined the code, and everything's looking great now in the scope of this PR. Ship it!
